recipes/sqlite3mc/all/test_package/main.c

```diff
@@ -0,0 +1,73 @@
+#include <sqlite3mc/sqlite3mc.h>
+#include <stdio.h>
+#include <string.h>
+
+int main(int argc, char *argv[])
+{
+    sqlite3 *db;
```

Comment:
Hi! Could we get a simplified version of this? It's not in CCI's policy to have extensive tests in the test_package, it should be as minimal as possible while testing the correct _package_ of the library :)

Reply:
> Hi! Could we get a simplified version of this? It's not in CCI's policy to have extensive tests in the test_package, it should be as minimal as possible while testing the correct _package_ of the library :)

Well, this **is** the most simple test I could think of. To verify that the package was build correctly it is necessary to create an encrypted database with some content and to access it with the right passphrase and a wrong passphrase afterwards. 

If that test is **too complex**, we don't need any test at all.

Reply:
I removed the configuration option `SQLITE_ENABLE_JSON1`, because this option is no longer used by SQLite. And I added `FreeBSD` to the os list (instead of replacing `Macos` by `FreeBSD`), because this seems to be the only logical modification.

Regarding the "extensive" test I have no idea what you expect. What could and/or should be tested, if not the proper build of the package?

Reply:
Hi,
we could remove the error handlings from the test,  use logical OR on the result during operations, and check it once in the end. That would make 35-40 lines shorter. Would that work?

Reply:
> we could remove the error handlings from the test, use logical OR on the result during operations, and check it once in the end. That would make it 35-40 lines shorter. Would that work?

It is certainly possible to reduce the number of lines of code by improving the error handling code (for example, by introducing a small function that checks the return code and emits an error message if necessary), but I'm not sure if that is what @RubenRBS intends.